### PR TITLE
Fix a DDP graph capture issue

### PR DESCRIPTION
**Pull request by yaochengji**
There're issues in torch/xla DDP. https://github.com/pytorch/xla/issues/8396. After investigation, one of the issues are caused by the incorrect graph capture caused by the combination of inplace-update and non aten op.

Changes:
1. Modify `TryGetXlaTensor` to make sure we can always have the most updated version of tensors
2. Modify the learning rate of ddp test from 1e-4 to 1e-1 as 1e-4 is too small to verify the DDP correctness
3. Remove the `gradient_as_bucket_view = True` requirement in all the docs and code
4. Add the test_inplace_update.py file, note that `test_non_aten_op_after_partial_update` cannot get passed before the `TryGetXlaTensor` patch

### CODE REVIEW

**Comment by yaochengji** [comment]
cc @qihqi @tengyifei

**Comment by jeffhataws** [comment]
@tengyifei  will you enable TPU testing on this PR?

**Comment by tengyifei** [comment]
@yaochengji could you force push a commit? I just added a `tpuci` label but we need to retrigger the jobs to run TPU tests.

**Review by tengyifei — APPROVED** [review]
Thanks!

**Comment by yaochengji** [comment]
@tengyifei thanks for your review. Do I still need to force push as it is already merged?

**Comment by tengyifei** [comment]
oops, i merged too soon :) looks like tip-of-tree build is green, so nothing needed on your part now!

**Comment by jeffhataws** [comment]
@yaochengji  will you add this change to https://github.com/pytorch/xla/issues/8455 for cherry-pick to 2.6? Let me know if I can help.

**Comment by yaochengji** [comment]
@jeffhataws it's added, thanks.
